chore: improve nx config support and caching

JeremyJonas opened the pull request:
### Improve NX project and target configurations
- enable PDKProject direct nx project config
- move default nx build target to workspace
- remove legacy nx override logic
- separate out aws-arch generate task and enable caching

### Improve aws-arch generator caching
- Replace "cache dir" with "hash based" temp dir to ensure unique empty dir while somewhat minimizing requests during dev
- Add `static/**` to generate input caching

JeremyJonas committed:
chore(nx-monorepo)!: improve nx project and target configs

- enable PDKProject direct nx project config
- move default nx build target to workspace
- remove legacy nx override logic
- separate out aws-arch generate task and enable caching

BREAKING CHANGE: modified nx-monorepo config settings for Nx
1. `WorkspaceConfig.nxConfig` type `NxConfig => Nx.WorkspaceConfig`
2. `overrideProjectTargets` removed in favor of `PDKProject.nx`
config to manage Nx project

See [PR 231](https://github.com/aws/aws-prototyping-sdk/pull/231)

JeremyJonas committed:
fix(aws-arch): fix generator caching

agdimech commented on packages/nx-monorepo/src/nx-monorepo.ts:
```diff
@@ -458,10 +345,7 @@ export class NxMonorepoProject extends TypeScriptProject {
-          ? { targets: this.targetOverrides[subProject.outdir] }
-          : undefined),
-          (manifest.name = subProject.name);
+        manifest.name = subProject.name;
```
why is this needed?

agdimech commented on packages/nx-monorepo/src/nx-types.ts:
```diff
@@ -0,0 +1,186 @@
+   *
+   * @see https://nx.dev/configuration/packagejson
+   */
+  export interface WorkspaceConfig {
```
Is this a breaking change given previously this was called NXConfig?

JeremyJonas replied:
Yes it is breaking, but only low-level config within nx-monorepo which should only be used by internals. As discussed I have updated the nx-monorepo README to include breaking details, along with commit message to denote such as well.

Given this is still v0, it is expected to break and we have now added sufficient messaging for potential external users.

agdimech reviewed (commented):
Overall it looks good and is much cleaner. Just had a few comments.

JeremyJonas commented:
@agdimech I have re-opened this stale/closed PR, which resolves build issues for https://github.com/aws/aws-prototyping-sdk/pull/266. 

I have rebased mainline and updated the commits as discussed - ready for your review.

JeremyJonas commented:
@agdimech After digging a bit deeper here, the real issue is the `aws-arch` generate script is simply not deterministic due to not being able to pin pricing manifest content that is fetch from the url. All the other sources are hashed urls, so deterministic between builds.

The only way I could make this deterministic is to commit the pricing-manifest.json file to the source code.

I also added a github workflow to run weekly to auto fetch the pricing and run build (which also updates the snapshots) - currently it requires manual approval since I want to see the snapshots since there is lots of auto mapping atm.

Hopefully this works well enough.

Commit for this new code here: https://github.com/aws/aws-prototyping-sdk/pull/231/commits/26112111c16b07e61a25991cd490cf7a8fb0fdc4

JeremyJonas committed:
fix(aws-arch): pricing manifest caching and automation

To support deterministic caching of aws-arch genoration of
aws pricing, we need to store the manifest locally since.

Creating github workflow to support weekly updates along with
snapshot updates and build verification - currently requires
manually approval.
